docs/migration-to-NRI.md

```diff
@@ -0,0 +1,182 @@
+# Prerequisities
+- Up and running CRI Resource Manager
+- One of the two supported policies in use: balloons or topology-aware.
+- For other policies a little bit more work is required and the policies need to be 'ported'. This can be done by just following the example of how balloons or topology-aware was converted.
+
+# Steps for an initial/basic migration test
+## Containerd 
+- Replace the containerd version in the system with 1.7 or newer version (NRI server not supported in older versions).
+
+- Get the Kubelet args:
+  ```console
+  systemctl cat kubelet <- Look for: EnvironmentFile=/.../kubeadm-flags.env
+  ```
+
+- Change the Kubelet container runtime endpoint to the containerd socket:
+    ```console 
+    $ sudo vim /.../kubeadm-flags.env
+    KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+    $ sudo vim /etc/sysconfig/kubelet 
+    KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/containerd/containerd.sock <- Remember this aswell
+
+    $ systemctl restart containerd
+    $ systemctl restart kubelet
+    ```
+
+- Edit the containerd config file and look for the section [plugins."io.containerd.nri.v1.nri"] and replace "disable = true" with "disable = false":
+  ```console
+  $ vim /etc/containerd/config.toml
+  ```
+  ```toml
+  [plugins."io.containerd.nri.v1.nri"]
+    disable = false
+    disable_connections = false
+    plugin_config_path = "/etc/nri/conf.d"
+    plugin_path = "/opt/nri/plugins"
+    plugin_registration_timeout = "5s"
+    plugin_request_timeout = "2s"
+    socket_path = "/var/run/nri/nri.sock"
+  ```
+
+## CRI-O
+- Replace the crio version in the system with 1.26.2 or newer.
+
+- Same thing with the Kubelet runtime endpoint, but change it to the crio socket and then restart the crio service.
+
+- Edit the crio config and look for the "[crio.nri]" section, replace "enable_nri = false" with "enable_nri = true" and uncomment the NRI config lines:
+  ```console
+  $ vim /etc/crio/crio.conf
+  ```
+
+  ```
+  # CRI-O NRI configuration.
+  [crio.nri]
+
+  # Globally enable or disable NRI.
+  enable_nri = true
+
+  # NRI configuration file to use.
+  nri_config_file = "/etc/nri/nri.conf"
+
+  # NRI socket to listen on.
+  nri_listen = "/var/run/nri/nri.sock"
+
+  # NRI plugin directory to use.
+  nri_plugin_dir = "/opt/nri/plugins"
+  ```
+
+## Build the policies
+  ```console
+    $ git clone https://github.com/jukkar/nri-resmgr.git
+    $ cd nri-resmgr
+    # Switch to a stable branch
+    $ git checkout preview
+    $ make
+    $ make images
+  ```
+
+## Apply required crds
+```console
+kubectl apply -f deployment/base/crds/noderesourcetopology_crd.yaml
+```
+
+## Import the image of the NRI plugin you want to run
+### Containerd
+  ```console
+  $ ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar
```

Comment:
Remove hash from:
`ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar`
because there will be no such file:
`ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-*.tar`

Reply:
If there are multiple tar files in the directory, then the *.tar would probably either fail or import wrong tar file. Best option here is to take the latest .tar file and import that.